apps/docs/src/guide/cli/commands.md

```diff
@@ -0,0 +1,203 @@
+1.  Generate types for them using `fuels-typegen` <sup>[2](#typegen)</sup>
+
+## `fuels deploy`
+
+```console
+npx fuels deploy
+```
+
+1. Deploy all Sway contracts under `workspace`
+1. Saves their deployed IDs to:
+   - _`./src/sway-programs-api/contracts.json`_
+
+```json
+{
+  "myContract1": "0x..",
+  "myContract2": "0x.."
+}
+```
+
+Use it when instantiating your contracts:
+
+```ts
+import { MyContract__factory } from "./sway-programs-api";
+import contractIds from './sway-programs-api/contracts.json'
+
+const contractId = contractIds.myContract1;
+const contract = MyContract__factory.connect(contractId, ..);
+```
+
+For a complete example, see:
+
+- [Using Generated Types](https://docs.fuel.network/docs/fuels-ts/abi-typegen/using-generated-types/)
+
+## `fuels dev`
+
+```console
+npx fuels dev
+```
+
+1. Runs `build` and `deploy` once at the start
+2. Watches your Sway workspace and repeats previous step on every change
+
+> _In `dev` mode, everytime you update a contract on your Forc `workspace`, we re-generate type definitions and factory classes for it, following your pre-configured [`output`](./config-file.md#output) directory. If it's part of another build system running in dev mode (i.e. `next dev`), you can expect it to re-build / auto-reload as well._
+
+## `fuels typegen`
+
+Manually generates type definitions and factory classes from ABI JSON files.
+
+```console
+npx fuels help typegen
+```
+
+```
+Usage: fuels typegen [options]
+
+Generate Typescript from Sway ABI JSON files
+
+Options:
+  -i, --inputs <path|glob...>  Input paths/globals to your Abi JSON files
+  -o, --output <dir>           Directory path for generated files
+  -c, --contract               Generate types for Contracts [default]
+  -s, --script                 Generate types for Scripts
+  -p, --predicate              Generate types for Predicates
+  -S, --silent                 Omit output messages
+  -h, --help                   Display help
+```
+
+For more info, check:
+
+- [Generating Types from ABI](https://docs.fuel.network/docs/fuels-ts/abi-typegen/generating-types-from-abi/)
+
+## `fuels versions`
+
+Check for version incompatibilities between your [Fuel Toolchain](#the-fuel-toolchain) component versions, matching them against the ones supported by the Typescript SDK version that you have.
+
+```console
+npx fuels versions
+```
+
+```
+You have all the right versions! ⚡
+┌───────────┬───────────┬─────────────────┐
+│           │ Supported │ Yours / System  │
+├───────────┼───────────┼─────────────────┤
+│ Forc      │ 0.30.0    │ 0.30.0          │
+├───────────┼───────────┼─────────────────┤
+│ Fuel-Core │ 0.14.0    │ 0.14.0          │
+└───────────┴───────────┴─────────────────┘
+```
+
+## The Fuel Toolchain
+
+This guide assumes you have [The Fuel Toolchain](https://docs.fuel.network/docs/sway/introduction/fuel_toolchain/) installed already.
```

Comment:
It would be, but I [changed](https://github.com/FuelLabs/fuels-ts/pull/1291#discussion_r1367206788) how the smart defaults selection is done, so I think it's good to keep it at the bottom as a reminder but not necessarily a requirement for the CLI to _work_. Users will see helpful log/warnings about this to reduce confusion.